src/frontend/interfaces/modal.interface.ts

```diff
@@ -0,0 +1,30 @@
+import { VariantType } from '@adminjs/design-system'
+
+export interface ConfirmModalData {
+  title?: string;
+  subTitle?: string;
+  variant?: VariantType;
+  label?: string;
+  icon?: string;
+  resourceId?: string;
+  type?: 'alert' | 'confirm';
+  onClick?: any;
+  confirmAction?: any;
+}
+
+export type ConfimOnClickHandler = () => void
+
+export interface ModalButton {
+  label?: string;
+  variant?: VariantType;
+  onClick?: ConfimOnClickHandler;
+}
+
+export type ModalFunction = (
+  data?: ConfirmModalData | null
+) => void
+
+export type ConfirmModalFunctions = {
+  openModal: ModalFunction,
+  closeModal: ModalFunction
```

Comment:
`closeModal` doesn't have `data` argument so this type is invalid

Reply:
fixed